src/main/java/bio/terra/service/snapshot/flight/create/SnapshotCreateFlight.java

```diff
@@ -292,6 +295,16 @@ public SnapshotCreateFlight(FlightMap inputParameters, Object applicationContext
     // The IAM code contains retries, so we don't make a retry rule here.
     addStep(new SnapshotAuthzIamStep(iamClient, snapshotService, snapshotReq, userReq, snapshotId));
 
+    // add data access control groups to the snapshot
+    List<String> dataAccessControlGroups = snapshotReq.getDataAccessControlGroups();
+    if (Objects.nonNull(dataAccessControlGroups) && !dataAccessControlGroups.isEmpty()) {
+      addStep(
+          new CreateSnapshotGroupConstraintPolicyStep(
```

Comment:
What do these two steps do?

Reply:
Good question - this is my best understanding: The first one creates a policy in TPS. Specifically, a policy that consists of the group constraints that represent the specified email groups. And the second one creates the auth domain in Sam which again represents the specified email groups. Someone correct me or add to this if it is incorrect/lacking.

Reply:
> they will need to be members of each of those groups, and be a member of the right snapshot policy

For the "create snapshot from request" API the plan is to add the DAC to the policy and then add the request user to the DAC. That way they'll get both the DAC members and the policy at the same time.